gh-141460: make RLock a class instead of a factory function (#142977)
python/cpython: Misc/NEWS.d/next/Library/2025-12-16-14-46-14.gh-issue-141460._44DYc.rst

> make threading.RLock a real class, not a factory function

Reviewer: ```suggestion
Make :class:`threading.RLock` a real class instead of a factory function.
```

> Make :class:`threading.RLock` a real class instead of a factory function.

Reviewer 2: Please mention also that the Python implementation has been removed.
